Pridanie Git modulu

garajm opened the pull request.

garajm committed:
Pridanie Git modulu do CMakeLists, vytvorenie GitFile, chybajuci include v LeapThread

garajm committed:
Vymazanie user konfiguracie

garajm committed:
Pridanie filepath a konstruktora triede ImportContext

garajm committed:
Pridanie moznosti na nacitanie git repozitara v CoreWindow

garajm committed:
Uprava Managera a ImporterContextu na import z gitu

garajm committed:
Triedy na vytvorenie tmp suboru a spracovanie system commandov

garajm committed:
Odstranenie pomocnych vypisov, pridana trieda GitFile

garajm committed:
Upravy GitFileLoadera

garajm committed:
Pridanie struktury pre diff blocky

garajm committed:
Pridanie struktury pre riadky z diff blocku

garajm committed:
Upravy pre aplikaciu DiffBlock a DiffBlockLine

garajm committed:
Pridanie triedy GitEvolutionGraph

garajm committed:
Upravy voci GitEvolutionGraph

garajm committed:
Dokoncenie metody loadGraphFromGit

garajm committed:
Pridanie resources pre EvolutionGraph

garajm committed:
Pridanie Node do grafu

garajm committed:
Pridanie triedy ReadEdgesStore

garajm committed:
Refactoring Node a OsgNode, prenesenie osg veci z Node do OsgNode

garajm committed:
Dokoncenie importera

garajm committed:
Uprava OsgNode

garajm committed:
Uprava zleho nazvu classy Data::Type

garajm committed:
Oprava nastavenia Edge cez settings

garajm committed:
Filter na GitFile podla extension

garajm committed:
Zmena std::map na QMap, korekcia funkcie contains a get

garajm committed:
Odstranenie zavislosti na Core v GitGraphImporter

garajm committed:
Pridanie GUI pre Evolution Graph

garajm committed:
Non Unity build upravy

garajm committed:
Pridanie vyhladania hrany podla mena

garajm committed:
Ziskanie suborov podla typu a extension

garajm committed:
Pridavanie novych uzlov po zmene verzie

kapecp committed:
[Build] Fix build on Linux

garajm committed:
Pridanie mapy vyskytu hrany v grafe a mensie upravy

garajm committed:
Oprava vyhladanie zmien podla extension

garajm committed:
Uprava nastavenia CoreGraph

garajm committed:
Oprava zleho nazvu

garajm committed:
Oprava nacitania iba verzii s .lua extension

garajm committed:
Pridanie triedy na upravovanie obsahu grafu podla verzii

garajm committed:
Uprava pre pouzitie novej triedy GitGraphUpdater

garajm committed:
Merge branch 'develop' of https://github.com/garajm/RealityNotFound into develop

garajm committed:
Pridanie komentov a popisovat atributor a metod

garajm committed:
Dokoncenie konfigurovatelneho extension filtra

garajm committed:
Pridanie obrazka pre zruseny Node

garajm committed:
Pridanie vypisu diff pre zvolene uzly, lifespan pre zrusene uzly, info o aktualnej verzii

garajm committed:
Odstranenie pomocnych vypisov, uprava zamknutia lifespan

garajm committed:
Doplnenie komentov a popisov

garajm committed:
Doplnenie komentov a popisov

garajm committed:
Astyle formatting

kapecp commented on src/Git/GitFileLoader.cpp:
```diff
@@ -0,0 +1,379 @@
+	if ( ok ) {
+		QProcess* process = new QProcess;
+		process->setStandardOutputFile( QDir::toNativeSeparators( tempFile.fileName() ) );
+		QString lCommand = QString( command );
```
je nutne vytvarat lCommand? nestaci pouzit command? pripadne len
QString lCommand = command

kapecp commented on src/Git/GitGraphUpdater.cpp:
```diff
@@ -0,0 +1,506 @@
+		}
+		switch ( gitFile->getType() ) {
+			// Ak bol pridany v dalsej verzii, tak ho pridame do grafu
+			case Git::GitType::ADDED :
```
switch neriesi enum Git::GitType::NONE - bud doplnit switch, alebo NONE je zbytocny
chyba default case

garajm replied:
Git::GitType::NONE sa nemôže pre triedu GitFile nastaviť. Zbytočný nie je využíva sa v triede Git FileDiffBlockLine. Je to riešené takto z dôvodu, že súčasný projekt nepodporuje enum class. Pridám tam default case.

kapecp commented on src/Git/GitGraphUpdater.cpp:
```diff
@@ -0,0 +1,506 @@
+			lPom += "/";
+		}
+
+		switch ( gitFile->getType() ) {
```
switch neriesi enum Git::GitType::NONE - bud doplnit switch, alebo NONE je zbytocny
chyba default case

kapecp commented on include/Git/GitGraphImporter.h:
```diff
@@ -0,0 +1,111 @@
+	 * std::auto_ptr<Importer::GraphOperations> graphOp
+	 * @brief Pointer to class, which sets up default edge and node type
+	 */
+	std::auto_ptr<Importer::GraphOperations> graphOp;
```
std::auto_ptr nahradit std::unique_ptr, ale treba skontrolovat podla:
https://stackoverflow.com/questions/3697686/why-is-auto-ptr-being-deprecated
https://stackoverflow.com/questions/3451099/stdauto-ptr-to-stdunique-ptr

kapecp commented on src/Git/GitFileLoader.cpp:
```diff
@@ -0,0 +1,379 @@
+				// o viacnasobne uvedenie identifikatoru a autora, vtedy precitame a zahodime. Ak nie je rovny 40, skonci vytvaranie verzie.
+				if ( A.indexIn( line ) != -1 ) {
+					line = line.replace( A, "projekt/" );
+					Git::GitFile* file = new Git::GitFile( line.mid( line.lastIndexOf( "/" ) + 1 ), line, Git::GitType::ADDED );
```
possible memmory leak: mame tu new, chyba delete
line sa vlozi do changedFiles, co je QList pointrov na  Git::GitFile
treba zmazat obsah toho changedFiles v destructore triedy GitVersion
vid:
http://doc.qt.io/qt-4.8/qtalgorithms.html#qDeleteAll-2
http://doc.qt.io/qt-4.8/qtalgorithms.html#qDeleteAll

podobne to je asi pre vsetky QList v ostatnych suboroch

kapecp commented on src/Git/GitFileLoader.cpp:
```diff
@@ -0,0 +1,379 @@
+
+	// Ak existuje na danej ceste git repozitar, tak vykonam command a vystup ulozim do temp suboru
+	if ( ok ) {
+		QProcess* process = new QProcess;
```
new, chyba delete - je nutne aby to bol pointer?

garajm replied:
Odstránil som pointer

kapecp committed:
Updated CPPLINT script - added folders of the new Git module

kapecp committed:
[Build] Added more GCC warnings, added note about -Weffc++ flag

kapecp committed:
Added some GCC diagnostic ignored pragmas to suppress too many warnings

kapecp committed:
Several fixes to the Git module

garajm committed:
Uprava pre pull request

garajm committed:
Merge zmien

garajm committed:
Stable build

garajm committed:
Odstranenie delete context

garajm committed:
Zmena adresarovej struktury, pridanie GitExport.h do tried GitLib

garajm committed:
Pridanie Repository namespace

garajm committed:
Vytvorenie GitLib

garajm committed:
Uprava ciest v include, pridanie git headerov do cmakelistu.

kapecp committed:
Fixed order of class member initialization

kapecp committed:
Removed direct class constructor calling

kapecp committed:
Fixed enum class vs. MSVC enum extension

kapecp committed:
[Build] More build tweaks for GitLib library

[+] added FLAGS_FOR_DEBUG2, so the library is compiled with -std=c++11
	(needed for enum class) and all common GCC warnings
[+] GitLib is cotired
[!] fixed dependecies between targets (including _unity targets)
[!] fixed installation of targets
